app/models/foreman_tasks/task/search.rb

```diff
@@ -20,7 +20,7 @@ def search_by_taxonomy(key, operator, value)
             foreman_tasks_links_taxonomy#{uniq_suffix}.resource_type = '#{resource_type}')
         SQL
         # Select only those tasks which either have the correct taxonomy or are not related to any
-        sql = "foreman_tasks_links_taxonomy#{uniq_suffix}.resource_id #{operator} ? OR foreman_tasks_links_taxonomy#{uniq_suffix}.resource_id IS NULL"
```

Comment:
wait, can't the operator be something else then `IN` would this work with `=` ? Could we add test for that as well?

Reply:
It can, but then the parentheses just group the expression within them, even if it is just a single value